internal/users/manager.go

```diff
@@ -415,13 +418,24 @@ func getActiveUsers(procDir string) (activeUsers map[string]struct{}, err error)
 	return activeUsers, nil
 }
 
-// GenerateID generates an integer number based on the provided string lowercased.
+// GenerateID deterministically generates an ID between from the given string, ignoring case. The ID is in the range
+// 60000 (the default value of UID_MAX, i.e. the maximum UID for regular users) and MaxInt32.
 func GenerateID(str string) int {
-	lowerCased := strings.ToLower(str)
-	var sum int
-	for i, c := range lowerCased {
-		// Multiplies the value of the rune (which is >0) by its index+1. Subtracts the index to add another layer of conflict prevention.
-		sum += int(c)*(i+1) - i
-	}
-	return (sum % (100000 - 65537)) + 65536 // Ensures that ID is between 65536 and 100000
+	const minID = 60000
+	const maxID = math.MaxInt32 // use MaxInt32 for 32-bit systems
+
+	str = strings.ToLower(str)
```

Comment:
This is what I mentioned on Monday: we should ensure that the check we had in authd that was removed in https://github.com/ubuntu/authd/pull/511 is added back in some ways.

I agree this should be the broker responsability. However, if we determine UID generation algorithm that relies on some property, like lowering the username, it means that we should consider in authd itself 2 users with different cases returning the same `ToLower()` strings as the same user.

It means then that the brokers can only be more retrictive than this, but not the opposite. Otherwise, it means that we returns 2 users that the broker consider different to the same UID in authd, which we should protect against.